### DISCUSSION TITLE
Edit form for multiple tables

### DISCUSSION THREAD

**Discussion by Rickmh** [Budibase/budibase#15058] [Help] [answered]
Hi,
With the help of this post (https://docs.budibase.com/docs/save-two-linked-tables-in-one-form) I've have created a form that allows me to insert data into two tables at the same time. 
Now I would like to create a form that allows me to edit the data in both tables within one form. I've tried the same approach as for the new form. However then the information of the second table is not loaded and also not saved. Any suggestions on how this can be achieved? Thanks.

**Comment by mikesealey** [comment, marked as answer]
Hey @Rickmh 

I've based the following on top of the app-export featured in the documentation that you posted. It's a little rough-around the edges, but you should hopefully get the picture.

In the Jobs table, I've added a button that runs 3 actions - The first two set some `_id`s in state (employee_id is fetched from the jobs assigned field), and the third navigates to a new page that contains a form.
![image](https://github.com/user-attachments/assets/0e6cb81c-abac-4d4a-a56c-a8c6d46699d8)

For debugging purposes, on the new form screen, I've added a headline to display the `_id`s passed in using state - you should delete these later, but it's helpful while you're still building.

The component tree starts to get a bit tricky here, but the reasons will become apparent. I've got a Data Provider fetching from the employee-table, filtering on the employee_id set in state. Inside that is a repeater (though there will only be one row returned)
![image](https://github.com/user-attachments/assets/e3e3ddd9-da0a-4894-b86c-92ce0b2a40d7)

Inside that repeater, add a form and a field group - the field group will go and fetch the fields for your form according to the schema you set your form.

*Also in that repeater* add another data provider, as a sibling of the form, nest another repeater, nest another form inside the repeater, and a field group inside that. These should all be set to Jobs. The Data Provider should be filtering the Jobs table based on the job `_id` 

Lastly, add a button to the field group of the inner-form. This will have access to both forms information. You can use it to save a row _twice_ one for each table.
![image](https://github.com/user-attachments/assets/41c0cd47-f85c-4577-86e3-108fe98e3c0b)

Hopefully with this approach you can apply it to your specific use-case.

**Reply by Rickmh** [reply]
Great thanks for your quick and elaborate response! I think that I understand. I'll give it a try and keep you posted.
